-1

Tested at: b2cfec73973efffe3333b545b8da570d7062ca24

CMSSW: CMSSW_11_1_X_2020-05-04-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9c5875/5998/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>++ scram build outputlog
Entering library rule at SimCalorimetry/HcalTrigPrimAlgos
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-05-04-2300/src/SimCalorimetry/HcalTrigPrimAlgos/src/HcalFinegrainBit.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-05-04-2300/src/SimCalorimetry/HcalTrigPrimAlgos/src/HcalFeatureHFEMBit.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-05-04-2300/src/SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-05-04-2300/src/SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc:816:59: error: expression result unused [-Werror,-Wunused-value]
        fgUpgradeMap_.insert(std::make_pair(id, element)).first;
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~
1 error generated.
gmake: *** [tmp/slc7_amd64_gcc820/src/SimCalorimetry/HcalTrigPrimAlgos/src/SimCalorimetryHcalTrigPrimAlgos/HcalTriggerPrimitiveAlgo.cc.o] Error 1
++ /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-05-04-2300/tmp/slc7_amd64_gcc820/cache/log/src
</pre>

--- a/SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc
+++ b/SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc
@@ -813,7 +813,7 @@ void HcalTriggerPrimitiveAlgo::addUpgradeFG(const HcalTrigTowerDetId& id,
       if (it == fgUpgradeMap_.end()) {
         FGUpgradeContainer element;
         element.resize(bits.size());
-        fgUpgradeMap_.insert(std::make_pair(id, element)).first;
+        fgUpgradeMap_.insert(std::make_pair(id, element));
       }
     }
 

